dev/tools/lib/roll_dev.dart

```diff
@@ -165,23 +167,26 @@ void main(List<String> args) {
 }
 
 String getFullTag() {
+  const String glob = '*.*.*-*.*.pre';
   return getGitOutput(
-    'describe --match *.*.*-dev.*.* --first-parent --long --tags',
+    'describe --match $glob --first-parent --long --tags',
     'obtain last released version number',
   );
 }
 
 Match parseFullTag(String version) {
-  final RegExp versionPattern = RegExp(r'^([0-9]+)\.([0-9]+)\.([0-9]+)-dev\.([0-9]+)\.([0-9]+)-([0-9]+)-g([a-f0-9]+)$');
+  final RegExp versionPattern = RegExp(
+    r'^(\d+)\.(\d+)\.(\d+)-(\d+)\.(\d+)\.pre-(\d+)-g([a-f0-9]+)$');
   return versionPattern.matchAsPrefix(version);
 }
 
 String getVersionFromParts(List<int> parts) {
   assert(parts.length == 5);
   final StringBuffer buf = StringBuffer()
     ..write(parts.take(3).join('.'))
-    ..write('-dev.')
-    ..write(parts.skip(3).join('.'));
+    ..write('-')
+    ..write(parts.skip(3).take(2).join('.'))
+    ..write('.pre');
```

Comment:
Here too so we don't have to guess what's getting skipped and taken.

Reply:
PTAL